state/postgresql/migrations.go

```diff
@@ -23,27 +23,33 @@ import (
 
 	"github.com/jackc/pgx/v5"
 
+	"github.com/dapr/components-contrib/internal/component/postgresql"
 	"github.com/dapr/kit/logger"
 )
 
 // Performs migrations for the database schema
 type migrations struct {
-	Logger            logger.Logger
-	Conn              pgxPoolConn
-	StateTableName    string
-	MetadataTableName string
+	logger            logger.Logger
+	stateTableName    string
+	metadataTableName string
 }
 
-// Perform the required migrations
-func (m *migrations) Perform(ctx context.Context) error {
+// performMigration the required migrations
+func performMigration(ctx context.Context, tx pgx.Tx, opts postgresql.EnsureTableOptions) error {
+	m := &migrations{
+		logger:            opts.Logger,
+		stateTableName:    opts.StateTableName,
+		metadataTableName: opts.MetadataTableName,
+	}
+
 	// Use an advisory lock (with an arbitrary number) to ensure that no one else is performing migrations at the same time
 	// This is the only way to also ensure we are not running multiple "CREATE TABLE IF NOT EXISTS" at the exact same time
 	// See: https://www.postgresql.org/message-id/CA+TgmoZAdYVtwBfp1FL2sMZbiHCWT4UPrzRLNnX1Nb30Ku3-gg@mail.gmail.com
 	const lockID = 42
 
 	// Long timeout here as this query may block
 	queryCtx, cancel := context.WithTimeout(ctx, time.Minute)
-	_, err := m.Conn.Exec(queryCtx, "SELECT pg_advisory_lock($1)", lockID)
+	_, err := tx.Exec(queryCtx, "SELECT pg_advisory_lock($1)", lockID)
```

Comment:
Please do not use a transaction here, but rather the actual connection. As explained in the link above, "CREATE TABLE" statements in Postgres do not respect transactions, so we must use a different kind of locking